pkg/p2p/streamtest/streamtest.go

```diff
@@ -156,6 +156,10 @@ func (s *stream) Close() error {
 	return nil
 }
 
+func (s *stream) FullClose() error {
```

Comment:
This method should behave as intended, to block until it reads an EOF from the other side.

Reply:
Actually, are we using `streamtest` for integration tests or unit test? I think that for unit tests, we don't need the actual functionality of waiting for the EOF. The side that calls `FullClose()` cares only if the error is returned or not, and waiting for the EOF is an underlying implementation. So no need to block, imo. Of course, we can introduce specific errors to simulate timeout and other cases as well.  Let's discuss this more on Tuesday.

Reply:
@janos I tried to mimic more of an actual libp2p implementation, but it looks like then all of our other implementations in streamtest would have to be implemented in the same way, and that looks like an overkill for now, imo. So, instead of that I went with a more simple idea, and it might be enough, because it still mimics the general behaviour of fullclose. Please, take a look at the last commit and let's discuss :)

Reply:
Cool, as we discussed, the approach is good. 👍

Reply:
@janos I have updated the code, and added test. The last test for multiple close and fullclose is a bit wild, lmnwyt :)
I did not touch handshake tests and fullclose mock, will do that in the separate PR. Also, I (or whoever get's a chance) would like to use separate PRs to add usage of fullclose in the actual protocols